### PR TITLE
Parser rebase

### PR DESCRIPTION
**Pull request by pxp928**
Signed-off-by: pxp928 [parth.psu@gmail.com](mailto:parth.psu@gmail.com)

Rebase the parser to be more pluggable for different document types

Closes: https://github.com/guacsec/guac/issues/83

### CODE REVIEW

File: pkg/ingestor/parser/common/types.go [hunk 1]
```diff
@@ -15,9 +15,20 @@
...
+	// Parse breaks out the document into the graph components
+	Parse(doc *processor.Document) error
+	// GetIdentities gets the identity node from the document if they exist
 	GetIdentities() []assembler.IdentityNode
+	// CreateNodes creates the GuacNode for the graph inputs
+	CreateNodes() []assembler.GuacNode
+	// CreateEdges creates the GuacEdges that form the relationship for the graph inputs
+	CreateEdges(foundIdentities []assembler.IdentityNode) []assembler.GuacEdge
+	// GetDocType returns the DocumentType that is being parsed
+	GetDocType() processor.DocumentType
```
Comment (lumjjb) [inline pkg/ingestor/parser/common/types.go:33]: not sure if this is super helpful? I think there may be a N:1 mapping of document types to parsers (e.g. ite6unknown and ite6slsa to one parser)

**Review by lumjjb — COMMENTED** [review]
minor nit, lgtm otherwise

File: pkg/ingestor/parser/common/types.go (hunk 1 above)
Comment (pxp928) [reply pkg/ingestor/parser/common/types.go]: Agreed. Removed.

**Review by mlieberman85 — APPROVED** [review]
LGTM

**Review by lumjjb — APPROVED** [review]

**Pull request merged**